File: PST/FUEL/src/FUEL/ExcelHandler.vb
Imports Microsoft.Office.Interop.Excel
Imports System
Imports System.Reflection

Namespace FUEL
    Public Class ExcelHandler
        ' Methods
        Public Shared Sub CloseWB(ByVal wb As Workbook)
            wb.Close(Missing.Value, Missing.Value, Missing.Value)
        End Sub

        Public Shared Function LoadExcel(ByVal FName As String) As Workbook
            Return New ApplicationClass().get_Workbooks.Open(FName, Missing.Value, True, Missing.Value, Missing.Value, Missing.Value, Missing.Value, Missing.Value, Missing.Value, Missing.Value, Missing.Value, Missing.Value, Missing.Value, Missing.Value, Missing.Value)
        End Function

        Public Shared Function ReadSheet(ByVal WB As Workbook, ByVal SheetNum As Integer) As Object(0 To .,0 To .)
            Return DirectCast(DirectCast(WB.get_Worksheets.get__Default(SheetNum), Worksheet).get_UsedRange.get_Value(DirectCast(10, XlRangeValueDataType)), Object(0 To .,0 To .)(,))
        End Function

    End Class
End Namespace

File: PST/FUEL Source Code/FUEL/ExcelHandler.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.ExcelHandler
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using Microsoft.Office.Interop.Excel;
using System.Diagnostics;
using System.Reflection;

namespace FUEL
{
  public class ExcelHandler
  {
    [DebuggerNonUserCode]
    public ExcelHandler()
    {
    }

    public static Workbook LoadExcel(string FName) => ((_Application) new ApplicationClass()).Workbooks.Open(FName, (object) Missing.Value, (object) true, (object) Missing.Value, (object) Missing.Value, (object) Missing.Value, (object) Missing.Value, (object) Missing.Value, (object) Missing.Value, (object) Missing.Value, (object) Missing.Value, (object) Missing.Value, (object) Missing.Value, (object) Missing.Value, (object) Missing.Value);

    public static object[,] ReadSheet(Workbook WB, int SheetNum) => (object[,]) ((_Worksheet) WB.Worksheets[(object) SheetNum]).UsedRange.get_Value((object) XlRangeValueDataType.xlRangeValueDefault);

    public static void CloseWB(Workbook wb) => wb.Close((object) Missing.Value, (object) Missing.Value, (object) Missing.Value);
  }
}
